qiskit/circuit/library/standard_gates/z.py

```diff
@@ -187,3 +188,79 @@ def __array__(self, dtype=None):
             return numpy.array(
                 [[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, -1, 0], [0, 0, 0, 1]], dtype=dtype
             )
+
+
+class CCZGate(ControlledGate):
+    r"""CCZ gate.
+
+    This is a symmetric gate.
+
+    Can be applied to a :class:`~qiskit.circuit.QuantumCircuit`
+    with the :meth:`~qiskit.circuit.QuantumCircuit.ccz` method.
+
+    **Circuit symbol:**
+
+    .. parsed-literal::
+
+        q_0: ─■─
+              │
+        q_1: ─■─
+              │
+        q_2: ─■─
+
+    **Matrix representation:**
+
+    .. math::
+
+        CCZ\ q_0, q_1, q_2 =
+            I \otimes I \otimes |0\rangle\langle 0| + CZ \otimes |1\rangle\langle 1| =
```

Comment:
Not quite sure this is right -- the Z term is applied only if both controls are 1 hence we should have a term of the form
```
Z \otimes |11\rangle\langle 11|
```
? I was expecting e.g.
```suggestion
            Z \otimes |11\rangle\langle 11| + I \otimes ... =
```
(writing out all other states than 11 is a bit annoying so I substituted by `...` 😉)

Reply:
Here I used the way [`CCXGate`](https://qiskit.org/documentation/stubs/qiskit.circuit.library.CCXGate.html) writes it. Doing the math it seems to work out.

Reply:
Alright, thanks for checking!